lib/crypto.js

```diff
@@ -518,9 +519,11 @@ function ECDH(curve) {
   this._handle = new binding.ECDH(curve);
 }
 
-exports.createECDH = function createECDH(curve) {
-  return new ECDH(curve);
-};
+if (hasEC()) {
+  exports.createECDH = function createECDH(curve) {
+    return new ECDH(curve);
+  };
+}
```

Comment:
I would much rather take the approach of throwing when these methods are called when EC is not enabled. The API surface area should not change and the error users would get if they tried to call these would not be helpful.

```js
exports.createECDH = function createECDH(curve) {
  if (ecUnavailable)
    throw new Error("Node.js is not compiled with openssl elliptic curve support.");
  return new ECDH(curve);
}
```